enacts/onset/maproom.py

```diff
@@ -768,6 +769,174 @@ def length_plots(
+        except KeyError:
+            error_fig = pingrid.error_fig(error_msg="Grid box out of data domain")
+            germ_sentence = ""
+            return error_fig, error_fig, tab_style
+        precip.load()
+        try:
+            onset_delta = calc.seasonal_onset_date(
+                precip,
+                int(search_start_day),
+                calc.strftimeb2int(search_start_month),
+                int(search_days),
+                int(wet_threshold),
+                int(running_days),
+                int(running_total),
+                int(min_rainy_days),
+                int(dry_days),
+                int(dry_spell),
+                time_coord="T",
+            )
+            isnan = np.isnan(onset_delta["onset_delta"]).mean()
+            if isnan == 1:
+                error_fig = pingrid.error_fig(error_msg="No onset dates were found")
+                return error_fig, error_fig, tab_style
+        except TypeError:
+            error_fig = pingrid.error_fig(
+                error_msg="Please ensure all onset input boxes are filled."
+            )
+            return error_fig, error_fig, tab_style
+        try:
+            soil_moisture = calc.water_balance(precip, 5, 60, 0)["soil_moisture"]
+            cess_delta = calc.seasonal_cess_date(
+                soil_moisture,
+                int(cess_start_day),
+                calc.strftimeb2int(cess_start_month),
+                int(cess_search_days),
+                int(cess_soil_moisture),
+                int(cess_dry_spell),
+                time_coord="T",
+            )
+            isnan = np.isnan(cess_delta["cess_delta"]).mean()
+            if isnan == 1:
+                error_fig = pingrid.error_fig(error_msg="No cessation dates were found")
+                return error_fig, error_fig, tab_style
+        except TypeError:
+            error_fig = pingrid.error_fig(
+                error_msg="Please ensure all cessation input boxes are filled"
+            )
+            return error_fig, error_fig, tab_style
+        if cess_delta["T"][0] < onset_delta["T"][0]:
+            cess_delta = cess_delta.isel({"T": slice(1, None)})
+        if cess_delta["T"].size != onset_delta["T"].size:
+            onset_delta = onset_delta.isel({"T": slice(None, -1)})
+        seasonal_edges = xr.concat(
+            [
+                (onset_delta["T"] + onset_delta["onset_delta"]).drop_vars("T"),
+                (cess_delta["T"] + cess_delta["cess_delta"]).drop_vars("T"),
+            ],
+            dim="edges"
+        ).where(lambda x: ~np.isnan(x).any(axis=0), drop=True).data.flatten("F")
+        try:
+            # Total seasonal rainfall between onset and cessation dates is
+            # every other sum over groups defined by onset and cessation dates
+            # as bins edges
+            seasonal_total = precip.groupby_bins(
+                "T",
+                seasonal_edges,
+                include_lowest=True
```

Comment:
One thing to double check is whether the inclusion is indeed inclusive for both edges. That's what I would want the way it is currently written.